src/components/Editor/components/MySteps/components/MyStepsList/index.js

```diff
@@ -7,34 +7,50 @@ import MyWork from '../../../../components/MyWork';
 import bootstrap from 'bootstrap/dist/css/bootstrap.min.css';
 import Locales from '../../../../../../strings';
 
+const CLEANUP_EXPLANATION = "(cleanup)";
+
 export default class MyStepsList extends Component {
+    constructor(props) {
+        super(props);
+
+        this.buildStep = this.buildStep.bind(this);
+    }
+
+    buildStep(i, value, explanation, isCleanup) {
+        var showTrash = false;
+        var showEdit = false;
+        if (i > 0 && !this.props.readOnly) {
+            showEdit = true;
+        }
+
+        if (i == this.props.solution.steps.length - 1 && this.props.solution.steps.length > 1 && !this.props.readOnly) {
+            showTrash = true;
+        }
+      
+        return <Step
+                key={`${isCleanup}-${i}`}
+                stepNumber={i + 1}
+                math={value}
+                annotation={explanation}
+                cleanup={isCleanup}
+                showEdit={showEdit && !isCleanup}
+                showTrash={showTrash && !isCleanup}
+                deleteStepCallback={this.props.deleteStepCallback}
+                editStepCallback={this.props.editStepCallback}
+                deleteStepsCallback={this.props.deleteStepsCallback}
+                readOnly={this.props.readOnly} />
+    }
+    
     render() {
-        var counter = 1;
-        let steps = this.props.solution.steps.map( (step, i) => {
-            var showTrash = false;
-            var showEdit = false;
-            if (i > 0 && !this.props.readOnly) {
-                showEdit = true;
+        var steps = [];
+        var counter = 0;
+        this.props.solution.steps.forEach((step) => {
+            steps.push(this.buildStep(counter++, step.stepValue, step.explanation, false));
```

Comment:
maybe oneliner?
steps.push(this.buildStep(step.cleanup ? counter : counter++, step.stepValue, step.explanation,step.cleanup));